GenericsHomework/Node.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+namespace GenericsHomework;
+
+
+public class Node<T>
+{
+
+    public Node<T> Next { get; private set; }
+    public T Data { get; }
+
+    public Node(T data)
+    {
+        Data = data;
+        Next = this;
+    }
+
+    public void Append(T value)
+    {
+        if(Exists(value))
+        {
+            throw new InvalidOperationException(nameof(value));
+        }   
+        Node<T> newNode = new(value);
+        Node<T> cur = this;
+
+        while (cur.Next != this)
+        {
+            cur = cur.Next;
+        }
+
+        newNode.Next = cur.Next;
+        cur.Next = newNode;
```

Comment:
```suggestion
  if (Exists(value)) throw new ArgumentException("This item already exists in the list");
  Node<T> newNode = new(value);
  newNode.Next = Next;
  Next = newNode;
  return newNode;
```
I think this can be simplified